internal/controller/atlasschema_controller.go

```diff
@@ -149,7 +157,7 @@ func (r *AtlasSchemaReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	// Starting area to handle the heavy jobs.
 	// Below this line is the main logic of the controller.
 	// ====================================================
-	if data.DevURL == "" {
+	if data.URL != nil && data.DevURL == "" {
```

Comment:
We don't support DevDB when using custom config right? Then, we need add an error here to report the case to the user.